### PR TITLE
Improve BSpline and CubicSpline Documentation

### LINKED ISSUE #15015: Improve document for spline functions
The documentation for the spline functions http://docs.mantidproject.org/nightly/fitfunctions/BSpline.html and http://docs.mantidproject.org/nightly/fitfunctions/CubicSpline.html currently give little information to why they are different.

As part of this issue:

* Explain when user may use CubicSpline and BSpline
* Explain all the attributes you can set with these functions
* Make the documention for these function easier to understand by non-expert, for instance by adding picture for each of these Spline functions with descriptive text

### PR DESCRIPTION
**Pull request by MattKing06**
The documentation for BSpline and CubicSpline fitting functions is lacking explanations of what these functions are and how they can be used in the sense of fitting.

**To test**
- Requires you to build qt-help pages
- Review Documentation of BSpline and CubicSpline
- Make sure Doc Tests pass

[Release Notes](http://www.mantidproject.org/ReleaseNotes_3_7_UI_Changes#Documentation)

Fixes #15015

### CODE REVIEW

File: docs/source/fitfunctions/CubicSpline.rst [hunk 1]
```diff
@@ -9,22 +9,90 @@ CubicSpline
...
 First and second derivatives from the spline can be calculated by using
 the derivative1D function.
 
-CubicSpline function takes a set of attributes and a set of parameters.
-The first attrbiute is 'n' which has integer type and sets the number of
-interpolation points. The parameter names have the form 'yi' where 'y'
-is letter 'y' and 'i' is the parameter's index starting from 0 and have
-the type double. Likewise, the attribute names have the form 'xi'.
+A CubicSpline is a polynomial function :math:`f(x)` of order 3, defined between an interval :math:`a \leqslant x \leqslant b`.
+When using CubicSplines for interpolation or for fitting, we essentially chain CubicSplines together so that each
```
Comment (ElliotAOram) [inline docs/source/fitfunctions/CubicSpline.rst]: Do you chain CubicSplines together or is it regular splines that are chained to make a Cubicspline?

File: docs/source/fitfunctions/CubicSpline.rst [hunk 2]
```diff
@@ -9,22 +9,90 @@ CubicSpline
...
+    :align: center
+    :height: 600px
+    :alt: fitting example using BSplines
+    
 
 .. attributes::
 
+   n;Integer;3;Number of breakpoints in Spline
+   x0;Double;\-;Position of first exterior breakpoint
+   x1;Double;\-;Position of the interior breakpoints
```
Comment (ElliotAOram) [inline docs/source/fitfunctions/CubicSpline.rst]: Should `breakpoints` be plural here? If so then perhaps the type is wrong?

**Comment by MattKing06** [comment]
@ElliotAOram As far as I understand we chain cubic splines together, However I could word this better such as "We chain together Basis-splines of order 3 together to create the cubic spline". 

I believe that is a typo, I'll fix it now and push back up with the changes.

**Comment by ElliotAOram** [comment]
**This PR was made before the changes to how we deal with merging code as such I will add in the comment that is normally auto generated**

Thanks for the changes. Here's a few things that we'll use to start the review process:

#### Main Reviewer ####

Please comment on the following:

##### Pull Request #####

- [ ] Does the title summarize the changes, i.e. it's not the automatically generated one?
- [ ] Is an issue link included/required, e.g. "Fixes #XXXX"? If there is an issue link do the changes look sensible and consistent with issue description?
- [ ] Are there adequate testing instructions?
- [ ] A link to the updated release notes has been provided or no release note updates were necessary?

##### Code Review #####

- [ ] Is the code of an acceptable quality?
  - [ ] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
  - [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
  - [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
  - [ ] Is user-facing documentation written in a user-friendly manner?
  - [ ] Has developer documentation been updated if required?

***

- Does everything look good? Comment with the ship it emoji but don't merge.


#### Second Reviewer (`@gatekeepers`) ####

- [ ] Have adequate tests, both success & failure cases, been performed?,
- [ ] Has the code been reviewed?,
- [ ] Ready to go? Hit `Merge`!

**Comment by ElliotAOram** [comment]
Changes look good and the test have all passed. Docs are much improved :shipit:

**Comment by FedeMPouzols** [comment]
:+1:

**Comment by FedeMPouzols** [comment]
I would be good to mention in the release note that the doc now provides concrete equations and examples (this has a general request from this year's meetings, around documentation of algorithms and functions).

**Pull request merged**